src/pages/patientView/clinicalInformation/PatientViewPageStore.ts

```diff
@@ -728,7 +729,21 @@ export class PatientViewPageStore {
     // set version 2 of the mutational signature as default
     @observable _selectedMutationalSignatureVersion: string =
         MutationalSignaturesVersion.V2;
+    // initialLoad to set _selectedMutationalSignatureVersion while loading the tab for the first time
+    @observable initialLoad: boolean = true;
     @computed get selectedMutationalSignatureVersion() {
+        if (
+            this.initialLoad &&
+            this.fetchAllMutationalSignatureData.isComplete
+        ) {
+            let versionPresent = retrieveMutationalSignatureVersionFromData(
+                this.fetchAllMutationalSignatureData.result.map(
+                    profile => profile.molecularProfileId
+                )
+            );
+            this.setMutationalSignaturesVersion(versionPresent);
```

Comment:
We should avoid setting another observable variable in the `@computed` component. It's because a computed component should only capture the changes of the variables it's monitoring.
We can do this:

1. Change `_selectedMutationalSignatureVersion` and `initialLoad` as described in my previous comments
2. In this computed components, return `version present` if `_selectedMutationalSignatureVersion === undefined` or return `_selectedMutationalSignatureVersion`.
The second step's logic is when  `_selectedMutationalSignatureVersion` is undefined, we know its initial load, and we can select the default available version from `version present`, if `_selectedMutationalSignatureVersion` has value, then that means users already set the version manually, then we can just return `_selectedMutationalSignatureVersion`.

Reply:
@TJMKuijpers this needs to be addressed